### ISSUE TITLE
init deploys with default profile if the selected profile is invalid

### ISSUE THREAD

**Issue by alex4108** [aws-amplify/amplify-cli#5777]
>**Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the 
[Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**
When creating a new project using `amplify init`, if the profile credentials defined in ~/.aws/credentials are invalid, the Amplify CLI does not catch this error and proceeds init with a (random?) profile

**Amplify CLI Version**
4.32.0

**To Reproduce**
1. Modify ~/.aws/credentials for the intended profile, modify key `aws_secret_access_key` to read `aws_secret-access_key`.
2. Validate aws cli is capable of handling the error
```
alex@DESKTOP-6H0TC0U:~/repos/it-inventory/inventory$ aws sts get-caller-identity --profile broken
Partial credentials found in shared-credentials-file, missing: aws_secret_access_key
```
3. Init a new amplify project
```
alex@DESKTOP-6H0TC0U:~/repos/it-inventory/inventory$ amplify init
Note: It is recommended to run this command from the root of your app directory
? Enter a name for the project inventory
? Enter a name for the environment dev
? Choose your default editor: Visual Studio Code
? Choose the type of app that you're building javascript
Please tell us about your project
? What javascript framework are you using vue
? Source Directory Path:  src
? Distribution Directory Path: dist
? Build Command:  npm run-script build
? Start Command: npm run-script serve
Using default provider  awscloudformation

For more information on AWS Profiles, see:
https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html

? Do you want to use an AWS profile? Yes
? Please choose the profile you want to use broken
Adding backend environment dev to AWS Amplify Console app: d11yelff4xoqvu
⠇ Initializing project in the cloud...

CREATE_IN_PROGRESS UnauthRole                   AWS::IAM::Role             Wed Nov 04 2020 20:51:22 GMT-0600 (Central Standard Time) Resource creation Initiated
CREATE_IN_PROGRESS AuthRole                     AWS::IAM::Role             Wed Nov 04 2020 20:51:22 GMT-0600 (Central Standard Time) Resource creation Initiated
CREATE_IN_PROGRESS UnauthRole                   AWS::IAM::Role             Wed Nov 04 2020 20:51:21 GMT-0600 (Central Standard Time)
CREATE_IN_PROGRESS DeploymentBucket             AWS::S3::Bucket            Wed Nov 04 2020 20:51:21 GMT-0600 (Central Standard Time)
CREATE_IN_PROGRESS AuthRole                     AWS::IAM::Role             Wed Nov 04 2020 20:51:21 GMT-0600 (Central Standard Time)
CREATE_IN_PROGRESS amplify-inventory-dev-205111 AWS::CloudFormation::Stack Wed Nov 04 2020 20:51:17 GMT-0600 (Central Standard Time) User Initiated
⠋ Initializing project in the cloud...

CREATE_IN_PROGRESS DeploymentBucket AWS::S3::Bucket Wed Nov 04 2020 20:51:23 GMT-0600 (Central Standard Time) Resource creation Initiated
⠦ Initializing project in the cloud...

CREATE_COMPLETE UnauthRole AWS::IAM::Role Wed Nov 04 2020 20:51:38 GMT-0600 (Central Standard Time)
CREATE_COMPLETE AuthRole   AWS::IAM::Role Wed Nov 04 2020 20:51:38 GMT-0600 (Central Standard Time)
⠏ Initializing project in the cloud...

CREATE_COMPLETE amplify-inventory-dev-205111 AWS::CloudFormation::Stack Wed Nov 04 2020 20:51:46 GMT-0600 (Central Standard Time)
CREATE_COMPLETE DeploymentBucket             AWS::S3::Bucket            Wed Nov 04 2020 20:51:44 GMT-0600 (Central Standard Time)
✔ Successfully created initial AWS cloud resources for deployments.
✔ Initialized provider successfully.
Initialized your environment successfully.

Your project has been successfully initialized and connected to the cloud!
```

After some digging, I determined by observing the ARNs listed in `amplify/cli.json` that the cloudformation template was deployed using a profile I did not select.

**Expected behavior**
Cloudformation template should deploy using indicated profile

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04 in WSLv2
 - Node Version. 14.4.0

**Comment by akshbhu** [comment]
Hi @alex4108 

Can you provide more details on this line : 

> Modify ~/.aws/credentials for the intended profile, modify key aws_secret_access_key to read aws_secret-access_key.

I just changed my ./aws/credentials profile to some random string and when doing ```amplify init``` , I selected the ```broken ``` profile and this gives me this error : `InvalidClientTokenId: The security token included in the request is invalid.`

**Comment by akshbhu** [comment]
Hi @alex4108 

Thanks for reporting it. I am marking this as bug.

**Comment by jhockett** [comment]
A fix for this has been merged (#7017) and released in v4.48.0.
